src/nat/agent/router_agent/agent.py

```diff
@@ -0,0 +1,332 @@
+
+                # Determine chosen branch based on agent response
+                if state.chosen_branch == "":
+                    for branch in self._branches:
+                        if branch.name.lower() in str(agent_response.content).lower():
+                            state.chosen_branch = branch.name
+                            if self.detailed_logs:
+                                logger.debug("%s Router Agent has chosen branch: %s", AGENT_LOG_PREFIX, branch.name)
+                            return state
+
+                # The agent failed to choose a branch
+                if state.chosen_branch == "":
+                    if attempt == self.max_router_retries:
+                        logger.error("%s Router Agent has empty chosen branch", AGENT_LOG_PREFIX)
+                        raise RuntimeError("Router Agent failed to choose a branch")
+                    logger.warning("%s Router Agent failed to choose a branch, retrying %d out of %d",
+                                   AGENT_LOG_PREFIX,
+                                   attempt,
+                                   self.max_router_retries)
+
+            except Exception as ex:
+                logger.error("%s Router Agent failed to call agent_node: %s", AGENT_LOG_PREFIX, ex)
+                raise
+
+        return state
+
+    async def branch_node(self, state: RouterAgentGraphState):
+        """Execute the selected branch with the forwarded message.
+
+        This method calls the tool/branch that was selected by the agent node
+        and processes the response.
+
+        Args:
+            state: The current state containing the chosen branch and message.
+
+        Returns:
+            RouterAgentGraphState: Updated state with the branch response.
+
+        Raises:
+            RuntimeError: If no branch was chosen or branch execution fails.
+            ValueError: If the requested tool is not found in the configuration.
+        """
+        logger.debug("%s Starting Router Agent Tool Node", AGENT_LOG_PREFIX)
+        try:
+            if state.chosen_branch == "":
+                logger.error("%s Router Agent has empty chosen branch", AGENT_LOG_PREFIX)
+                raise RuntimeError("Router Agent failed to choose a branch")
+            requested_branch = self._get_branch(state.chosen_branch)
+            if not requested_branch:
+                logger.error("%s Router Agent wants to call tool %s but it is not in the config file",
+                             AGENT_LOG_PREFIX,
+                             state.chosen_branch)
+                raise ValueError("Tool not found in config file")
+
+            branch_input = state.forward_message.content
+            branch_response = await self._call_tool(requested_branch, branch_input)
+            state.messages += [branch_response]
+            if self.detailed_logs:
+                self._log_tool_response(requested_branch.name, state.messages[-1].content, branch_response.content)
```

Comment:
Ah, I think this is why :) (You added the branch_response already, so you want the second-from-last)
```suggestion
                self._log_tool_response(requested_branch.name, state.messages[-2].content, branch_response.content)
```